clients/include/testing_csx2dense.hpp

```diff
@@ -0,0 +1,396 @@
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       descr,
+                       (const T*)nullptr,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, nullptr, d_csx_col_row_ind, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, d_csx_row_col_ptr, nullptr, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)nullptr,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    //
+    // Testing invalid size on M
+    //
+    status = csx2dense(handle,
+                       -1,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
+    //
+    // Testing invalid size on N
+    //
+    status = csx2dense(handle,
+                       M,
+                       -1,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
+    //
+    // Testing invalid size on LD
+    //
+    status = csx2dense(handle,
+                       M,
+                       -1,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       M - 1);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
```

Comment:
Should be verify_hipsparse_status_invalid_size?

Reply:
Well, good catch, will fix use the right routine.
Behind the scene, v_h_s_invalid_size and v_h_s_invalid_pointer check for the same hip status invalid value. This is what the hip status proposes. I don't necessarily want to enhance the status right now.